### ISSUE TITLE
Milvus not to response after queryed for a period of time

### ISSUE THREAD

**Issue by linven0721** [milvus-io/milvus#4459]
使用milvus-0.10.4 gpu版本docker部署，底库数据量几十条，使用50并发查询，在进行了十几万次查询后，请求无响应，单独进行一次查询后台日志如下：
![微信图片_20201214103237](https://user-images.githubusercontent.com/30649209/102033971-d550f080-3df7-11eb-812c-e0eceab700e5.png)

请问这种问题怎么解决？

**Comment by yhmo** [comment]
Could relate to #4378 Multi-threads to call load_collection() and search() cause Milvus hang.
Did you call load_collection() or get_entity_by_id() during searching?

**Comment by linven0721** [comment]
> Could relate to #4378 Multi-threads to call load_collection() and search() cause Milvus hang.
> Did you call load_collection() or get_entity_by_id() during searching?

我的调用流程是这样的：
1.第一次会根据collectionName去调用hasCollection函数查询集合是否存在，如果存在的话，会查询该集合的所有partition，将查询结果保存在内存中，后面的所有查询会直接使用内存中的数据。   
2.查询是带partition tags进行查询所有分区的。

**Comment by yhmo** [comment]
I **could not reproduce** this issue with this script:

```
import sys
import time
import numpy as np
import random
import threading

from milvus import Milvus, IndexType, Status
from milvus.client.types import MetricType

SERVER_ADDR = "127.0.0.1"
SERVER_PORT = '19530'
COLLECTION_NAME = "TEST"
COLLECTION_DIMENSION = 128
COLLECTION_PARAM = {'collection_name': COLLECTION_NAME,
                    'dimension': COLLECTION_DIMENSION,
                    'index_file_size': 1024,
                    'metric_type':MetricType.L2
                    }


NQ = 1
TOPK = 10
SEARCH_PARAM = {'nprobe': 32}

MILVUS = Milvus(host=SERVER_ADDR, port=SERVER_PORT)

def handle_status(status):
    if status.code != Status.SUCCESS:
        print(status)
        sys.exit(2)

def create_collection():
    print(COLLECTION_PARAM)
    status = MILVUS.create_collection(COLLECTION_PARAM)
    print(COLLECTION_NAME, "created")

def drop_collection():
    print("drop collection")
    status = MILVUS.drop_collection(collection_name=COLLECTION_NAME)
    print(status)
    return status

def count_collection():
    status, row_count = MILVUS.count_entities(collection_name=COLLECTION_NAME)
    print(COLLECTION_NAME, "row count =", row_count)

def flush_collection():
    status = MILVUS.flush(collection_name_array=[COLLECTION_NAME])
    print(COLLECTION_NAME, "flushed")

def create_partitions(tag):
    print("create partitions...")
    status = MILVUS.create_partition(COLLECTION_NAME, tag)
    handle_status(status)
    return status


def gen_vec_list(nb, seed=np.random.RandomState(1234)):
    xb = seed.rand(nb, COLLECTION_DIMENSION).astype("float32")
    vec_list = xb.tolist()
    return vec_list

def add_vec_to_milvus(vec_list, vector_ids=None, tag=None):
    status, ids = MILVUS.insert(collection_name=COLLECTION_NAME, records=vec_list, ids=vector_ids, partition_tag=tag)
    handle_status(status=status)
    #print(ids)
    return ids

def print_result(results):
    print("query results for each vector:")
    for each_query in results:
        print("topk result:")
        for each_score in each_query:
            print("id = ", each_score.id, " distance = ", each_score.distance)

def search(tags):
    for i in range(100000):
        if i%100 == 0:
            print(threading.current_thread().name, i)
        query_vectors = gen_vec_list(NQ)
        status, results = MILVUS.search(collection_name=COLLECTION_NAME, query_records=query_vectors,
                                        partition_tags=tags, top_k=TOPK, params=SEARCH_PARAM)
        handle_status(status)
    # print_result(results)

def main(argv):
    drop_collection()
    create_collection()
    for i in range(10):
        tag = 'p_' + str(i)
        create_partitions(tag=tag)

    vec_list = gen_vec_list(5)
    add_vec_to_milvus(vec_list=vec_list)

    repeat = 100
    for i in range(repeat):
        print("No.", i)
        tag = 'p_' + str(i%10)
        vec_list = gen_vec_list(1)
        add_vec_to_milvus(vec_list=vec_list, tag=tag)

    flush_collection()
    count_collection()
    # create_index()

    status, parts = MILVUS.list_partitions(collection_name=COLLECTION_NAME)
    print(parts)
    tags = []
    for part in parts:
        tags.append(part.tag)
    print(tags)

    THREAD_COUNT = 50
    threads = []
    for k in range(THREAD_COUNT):
        x = threading.Thread(target=search, args=(tags,))
        threads.append(x)
        x.start()

    for th in threads:
        th.join()


if __name__ == "__main__":
    main(sys.argv[1:])
```

**Comment by yhmo** [comment]
@linven0721 
The upper script create a collection with 10 partitions, each partitions has 10 vectors, and run search() for 5000000 times. It could not reproduce your problem. Is anything missed?

**Comment by yhmo** [comment]
From this issue, I found a bug of search(), I create an issue for the new bug: [#4484](https://github.com/milvus-io/milvus/issues/4484) Milvus only search default partition if search parameter 'partition_tags' contains '_default'

**Comment by shengjun1985** [comment]
It should be fixed in 0.10.5.
